base/boot.jl

```diff
@@ -448,8 +448,7 @@ end
 mutable struct WeakRef
     value
     WeakRef() = WeakRef(nothing)
-    WeakRef(@nospecialize(v)) = ccall(:jl_gc_new_weakref_th, Ref{WeakRef},
-                                      (Ptr{Cvoid}, Any), getptls(), v)
+    WeakRef(@nospecialize(v)) = ccall(:jl_gc_new_weakref, Ref{WeakRef}, (Any, ), v)
```

Comment:
Generally the trend has been the other way. Ideally we want expose `current_task()` such that the compiler can merge the lookups within a function. Then the ptls load is a load from that cached pointer instead of going through a thread-local lookup.

I don't know if we are doing enough coalescing for this to matter here, but that is the overarchiygoal with passing ptls in.

Reply:
See https://github.com/JuliaLang/julia/pull/17116 for the history and motivation.

As I said I am unsure if LLVM will do this optimization for us, but there is a draft of a GVN pass that we will be able to teach that `current_task()` is meet able across functions.

Reply:
Thanks Valentin!

There seems to be performance considerations with using new_weakref_th then.

I'll be closing the PR in light of that, and we can come back to this if needed.